src/Poseidon/Package.hs

```diff
@@ -809,41 +769,72 @@ writePoseidonPackage (PoseidonPackage baseDir ver tit des con pacVer mod_ geno j
-}
-
-getAllGroupInfo :: [PoseidonPackage] -> [GroupInfo]
+packagesToPackageInfos :: (MonadThrow m) => [PoseidonPackage] -> m [PackageInfo]
+packagesToPackageInfos pacs = do
+    forM pacs $ \pac -> do
+        isLatest <- isLatestInCollection pacs pac
+        return $ PackageInfo {
+            pPac           = posPacNameAndVersion pac,
+            pIsLatest      = isLatest,
+            pPosVersion    = posPacPoseidonVersion pac,
+            pDescription   = posPacDescription pac,
+            pLastModified  = posPacLastModified pac,
+            pNrIndividuals = (length . getJannoRowsFromPac) pac
+        }
+
+getAllGroupInfo :: (MonadThrow m) => [PoseidonPackage] -> m [GroupInfo]
 getAllGroupInfo packages = do
     let individualInfoUnnested = do
             pac <- packages
             jannoRow <- getJannoRowsFromPac pac
             let groups = getJannoList . jGroupName $ jannoRow
-                pacName = posPacTitle pac
-                pacVersion = posPacPackageVersion pac
-            [(g, PacNameAndVersion (pacName, pacVersion)) | g <- groups]
-    group_ <- group . sort $ individualInfoUnnested
-    let groupName     = head . map fst $ group_
-        groupPacs     = head . map snd $ group_
-        groupNrInds   = length group_
-    return $ GroupInfo groupName groupPacs groupNrInds
-
-getExtendedIndividualInfo :: [PoseidonPackage] -> [String] -> [ExtendedIndividualInfo]
-getExtendedIndividualInfo allPackages additionalJannoColumns = do
-    pac <- allPackages
+            [(g, makePacNameAndVersion pac) | g <- groups]
+    -- loop over pairs of groups and pacNames
+    forM ((group . sort) individualInfoUnnested) $ \group_ -> do
+        let groupName   = head . map fst $ group_
+            groupPac    = head . map snd $ group_
+            groupNrInds = length group_
+        isLatest <- isLatestInCollection (map makePacNameAndVersion packages) groupPac
+        return $ GroupInfo groupName groupPac isLatest groupNrInds
+
+getJointIndividualInfo :: [PoseidonPackage] -> [IndividualInfo]
+getJointIndividualInfo packages = do
+    pac <- packages
     jannoRow <- getJannoRowsFromPac pac
-    let name = jPoseidonID jannoRow
-        groups = getJannoList . jGroupName $ jannoRow
-        pacName = posPacTitle pac
-        version = posPacPackageVersion pac
-        additionalColumnEntries = case additionalJannoColumns of
-            [] -> []
-            colNames -> [(k, BSC.unpack <$> toNamedRecord jannoRow HM.!? BSC.pack k) | k <- colNames]
-    return $ ExtendedIndividualInfo name groups pacName version additionalColumnEntries
+    return $ IndividualInfo
+        (jPoseidonID jannoRow)
+        ((getJannoList . jGroupName) jannoRow)
+        (makePacNameAndVersion pac)
 
+getExtendedIndividualInfo :: (MonadThrow m) => [PoseidonPackage] -> [String] -> m [ExtendedIndividualInfo]
+getExtendedIndividualInfo allPackages additionalJannoColumns = do
+    concatForM allPackages $ \pac -> do
```

Comment:
This definitely is much easier and shorter with sequence and list-monad. I want to make the change.

Reply:
OK, I've made the change. Check out my last commit, I hope you agree.

Reply:
Fascinating!